online_meetings/bluejeans_report.py

```diff
@@ -33,141 +36,175 @@
-logger.info(payload)
-headers = {
-  'Accept': 'application/json',
-  'Content-Type': 'application/json'
-}
-
-token_expiration_time = datetime.now()
-response = requests.request("POST", url, headers=headers, data=payload).text.encode('utf8')
-token_json = json.loads(response)
-token = token_json["access_token"]
-logger.info(token)
-# the expiration time in seconds
-expires_in_sec = token_json["expires_in"]
-# minus for one minute, for safety check
-expires_in_sec = expires_in_sec - 60
-token_expiration_time = datetime.now() + timedelta(seconds=expires_in_sec)
-logger.info(f"{token} expires on: {token_expiration_time}")
-
-payload = {}
-headers = {
-  'Accept': 'application/json',
-  'Content-Type': 'application/json',
-  'Authorization': f'Bearer {token}'
-}
-
-page_size = 1000
-total_records = 0
-
-total_df = pd.DataFrame()
-
-# TODO: the earliest time is hardcoded now
-early_date = dateparser.parse(ENV.get('ZOOM_EARLIEST_FROM', '2020-04-02')).date()
-
-# Only use the date as a paramter
-# Loop until yesterday (this will go until now() -1)
-for i in range((datetime.now(pytz.timezone('America/Detroit')).date() - early_date).days):
-    param_date = early_date + timedelta(days=i)
-    start = datetime(param_date.year, param_date.month, param_date.day, 0, 0, 0)
-    end = datetime(param_date.year, param_date.month, param_date.day, 23, 59, 59)
-    logger.info("start")
-    logger.info(start.astimezone().isoformat())
-    logger.info("end")
-    logger.info(end.astimezone().isoformat())
-    
-    # post the report request job
-    url = "https://indigo-api.bluejeans.com/v1/enterprise/" + str(ENV.get('BLUEJEANS_ENTERPRISE_ID')) + "/indigo/reportJob/meetings?filter=\
-    [{\"type\":\"date\",\"comparison\":\"lt\",\"value\":\"" + end.astimezone().isoformat() + "\",\"field\":\"end_time\"},\
-    {\"type\":\"date\",\"comparison\":\"lt\",\"value\":\"" + end.astimezone().isoformat() + "\",\"field\":\"start_time\"},\
-    {\"type\":\"date\",\"comparison\":\"gt\",\"value\":\"" + start.astimezone().isoformat() + "\",\"field\":\"start_time\"}]\
-    &fileName=meetings_30th-Mar-2020_30th-Mar-2020&userid=" + str(ENV.get("BLUEJEANS_USER_ID")) + "&app_name=command_center"
+# output settings
+CREATE_CSVS = ENV.get('CREATE_CSVS', False)
+INVENTORY_DB = ENV['INVENTORY_DB']
+logger.info(INVENTORY_DB)
+APPEND_TABLE_NAMES = ENV.get('APPEND_TABLE_NAMES', ['job_run', 'data_source_status'])
+
+def run_bluejeans_report() -> Sequence[Dict[str, Union[str, pd.Timestamp]]]:
+    logger.info("run_bluejeans_report")
+
+    # Record data source info for Canvas API
+    bluejeans_data_source = {
+        'data_source_name': 'BLUEJEANS_API',
+        'data_updated_at': pd.to_datetime(time.time(), unit='s', utc=True)
+    }
+
+    url = "https://api.bluejeans.com/oauth2/token?Client"
+
+    payload = "{ \"grant_type\": \"client_credentials\", \"client_id\": \"Metrics_BJN\", \"client_secret\": \"" + ENV.get('BLUEJEANS_CLIENT_SECRET') + "\"}"
```

Comment:
The backward slashed around double quotes around the payload variable attributes are little glary to the eye. can you make it a dictionary and convert it to JSON?